### PR TITLE
Added 3D Mobject :class:`~.Dot3D`

### PR DESCRIPTION
**Pull request by Darylgolden**
## Motivation
To add spherical, three dimensional analogues of `Dot` and `SmallDot`.

## Overview / Explanation for Changes
Added 3D Mobjects: `Dot3D`

## Oneline Summary of Changes

```
- Added new feature ... / Fixed bug ... / ... (:pr:`PR NUMBER HERE`)
```

## Testing Status
Added tests for each Mobject. Tests pass locally.

## Acknowledgements
- [x] I have read the [Contributing Guidelines](https://docs.manim.community/en/latest/contributing.html)

## Reviewer Checklist
- [ ] Newly added functions/classes are either private or have a docstring
- [ ] Newly added functions/classes have [tests](https://github.com/ManimCommunity/manim/wiki/Testing) added and (optional) examples in the docs
- [ ] Newly added documentation builds, looks correctly formatted, and adds no additional build warnings
- [ ] The oneline summary has been included [in the wiki](https://github.com/ManimCommunity/manim/wiki/Changelog-for-next-release)

### CODE REVIEW

**Comment by WampyCakes** [comment]
I'm on board with adding a 3D Dot, though I think adding SmallDot3D is unnecessary. As the only difference between them is size and not shape, I think accepting the size in the constructor is sufficient. I'd also like to see a doc example, perhaps an image of a Dot3D on a set of axes? Otherwise, this looks good to me.

**Comment by Darylgolden** [comment]
> 
> 
> I'm on board with adding a 3D Dot, though I think adding SmallDot3D is unnecessary. As the only difference between them is size and not shape, I think accepting the size in the constructor is sufficient. I'd also like to see a doc example, perhaps an image of a Dot3D on a set of axes? Otherwise, this looks good to me.

I agree, but I added SmallDot3D as an analogue to SmallDot. I can remove it if necessary.

**Comment by AntonBallmaier** [comment]
I feel like resolution=(12, 24) is a bit overkill for a dot. Maybe reduce it to lower values.

**Comment by WampyCakes** [comment]
> I agree, but I added SmallDot3D as an analogue to SmallDot. I can remove it if necessary.

I understand the reasoning, but what's the limiting factor? Should there also be a BigDot or an ExtraSmallDot? IMO dot is good enough

**Comment by Darylgolden** [comment]
> 
> 
> > I agree, but I added SmallDot3D as an analogue to SmallDot. I can remove it if necessary.
> 
> I understand the reasoning, but what's the limiting factor? Should there also be a BigDot or an ExtraSmallDot? IMO dot is good enough

I understand, but to me it doesn't make sense to have `Dot` and `SmallDot` but only `Dot3D`. Either both `SmallDot` and `SmallDot3D` should be removed, or they should be both included.

**Comment by WampyCakes** [comment]
SmallDot probably should be removed imo

**Comment by Darylgolden** [comment]
Per the poll results in Discord, I have removed `SmallDot3D`. A separate PR should be made to remove `SmallDot` too.

File: manim/mobject/three_dimensions.py [hunk 1]
```diff
@@ -160,6 +161,20 @@ def func(
         return np.array([np.cos(v) * np.sin(u), np.sin(v) * np.sin(u), np.cos(u)])
 
 
+class Dot3D(Sphere):
+    """A spherical dot.
+
+    Parameters
+    --------
+    radius : :class:`float`
```
Comment (jsonvillanueva) [inline manim/mobject/three_dimensions.py:169]:
```suggestion
    radius : :class:`float`, optional
```

File: manim/mobject/three_dimensions.py [hunk 2]
```diff
@@ -160,6 +161,20 @@ def func(
...
+        The radius of the dot.
+    """
```
Comment (jsonvillanueva) [inline manim/mobject/three_dimensions.py:171]:
```suggestion
    color : :class:`~.Colors`, optional
        The color of the :class:`Dot3D`
```

**Review by jsonvillanueva — CHANGES_REQUESTED** [review]
Now that you can correctly render sphinx documentation, please have a look at our wiki on [documentation](https://github.com/ManimCommunity/manim/wiki/Documentation-guidelines-(WIP)) for future documentation. It'd be helpful to have a code example using our manim directive, but this isn't strictly necessary.

**Review by jsonvillanueva — APPROVED** [review]
LGTM! Although personally, I'm not too found of the name Dot3D, if it's what the community wants, then sure.

**Comment by Darylgolden** [comment]
Hmm, any suggestions for alternative names then?

**Comment by jsonvillanueva** [comment]
> 
> 
> Hmm, any suggestions for alternative names then?

Not really, it's just something I thought I'd mention. In plain English, a `Dot`, or "point", makes sense to be a *small* `Circle`... but you don't really call a *small* `Sphere` a "Dot3D". Other Spherical items include: balls, globes, orbs and well stars/suns if we're stretching the meaning of "perfect" Spheres, but aren't *that* good of options.

**Comment by WampyCakes** [comment]
I agree with @jsonvillanueva, though I think the reasonable name choices are limited. As this is just a small sphere, if you were to name it `SmallSphere` it wouldn't even make sense to include. I think the upside to `Dot3D` is it captures the essence of its use.

**Comment by WampyCakes** [comment]
I added an example of Dot3D and accidentally pushed it straight here. I meant to make a PR to your branch 😬

**Comment by WampyCakes** [comment]
I went ahead and pushed the black update now though.

**Review by WampyCakes — APPROVED** [review]
